datafusion/core/src/datasource/listing/table.rs

```diff
@@ -1092,6 +1092,95 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn load_table_stats_with_session_level_cache() -> Result<()> {
+        let testdata = crate::test_util::parquet_test_data();
```

Comment:
Add test to check cache share in session level

Reply:
Nice -- thank you. Since this is an end to end test, I recommend moving it to somewhere in `core_integration`:  datafusion/core/tests/core_integration.rs perhaps